lib/set.js

```diff
@@ -0,0 +1,112 @@
+'use strict';
+
+const Ref = require('./ref');
+
+const Set = class {
```

Comment:
@WesTyler oh my, sorry, I was thinking module.exports = class, just bypassing the local constant.